File: Algorithms/Implementation/Jumping on the Clouds.php
<?php

/*
 * Complete the 'jumpingOnClouds' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER_ARRAY c as parameter.
 */

function jumpingOnClouds($c) {
    # Write your code here
    $jumps = 0;
    $i = 0;
    while ($i < count($c) - 1) {
        if ($i + 2 < count($c) && $c[$i + 2] == 0) {
            $i += 2;
            $jumps++;
        }
        else {
            $i++;
            $jumps++;
        }
    }
    return $jumps;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));
$c_temp = rtrim(fgets(STDIN));
$c = array_map('intval', preg_split('/ /', $c_temp, -1, PREG_SPLIT_NO_EMPTY));
$result = jumpingOnClouds($c);

fwrite($fptr, $result . "\n");
fclose($fptr);

File: Algorithms/Implementation/Jumping on the Clouds.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'jumpingOnClouds' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER_ARRAY c as parameter.
     */

    public static int jumpingOnClouds(List<int> c)
    {
        int jumps = 0;
        int i = 0;
        while (i < c.Count - 1)
        {
            if (i + 2 < c.Count && c[i + 2] == 0)
            {
                i += 2;
                jumps++;
            }
            else
            {
                i++;
                jumps++;
            }
        }
        return jumps;
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> c = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(cTemp => Convert.ToInt32(cTemp)).ToList();

        int result = Result.jumpingOnClouds(c);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}
